src/plugins/discourse/references.js

```diff
@@ -23,3 +48,37 @@ export function parseLinks(cookedHtml: string): Hyperlink[] {
   parser.end();
   return links;
 }
+
+export function linksToReferences(
+  links: $ReadOnlyArray<Hyperlink>
+): DiscourseReference[] {
+  const server = "(https://[\\w.-]+)";
+  const topic = `${server}/t/[\\w-]+/(\\d+)`;
+  const post = `${topic}/(\\d+)`;
+  const params = "(?:\\?[\\w-=]+)?";
```

Comment:
Perhaps this needs a snapshot test from a corpus of real-world links.
Especially interesting would be to know if this URL scheme has changed between discourse versions.

Reply:
Added a snapshot test with a few links from forums I know of. None of them are super weird... if you find weirder links somewhere, feel free to add them.